source/develop/release-management/features/storage/cinderlib-integration.html.md

```diff
@@ -0,0 +1,144 @@
+
+
+### Owner
+
+*   Fred Rolland (<frolland@redhat.com>)
+*   Benny Zlotnik (<bzlotnik@redhat.com>)
+*   Eyal Shenitzky (<eshenitz@redhat.com>)
+
+### Detailed Description
+
+Cinder Library [1] provides a Python library that give the possibility to use the Cinder drivers directly without the need of a full Openstack/Cinder deployment supporting over 80 storage drivers.
+
+- The user is able to configure a Storage Domain with the name of the Cinder driver and a set of parameters needed by the driver implementation.
+- The user is able to create/delete disks on that Storage Domain.
+- The user is able to create/delete snapshots of the disks created on that Storage Domain.
+- The user is able to run VMs with disks created on that Storage Domain.
+
+
+### Benefit to oVirt
+
+With this feature the user will be able to use any storage vendor supported in Cinder (over 80 storage drivers) and beneficiate from the offloading features the storage vendors has implemented.
+
+### User Experience
+
+On the UX side, the storage flows are without any changes, except for the new Storage Domain type, where the user will have to provide a set of parameters needed by the driver (name/value pairs).
+Sensitive fields, such as passwords, will be separated from others and masked in the GUI. These fields will be stored encrypted in the DB.
+
+For example:
+- volume_driver='cinder.volume.drivers.dell_emc.xtremio.XtremIOISCSIDriver'
+- san_ip='10.10.10.1'
+- xtremio_cluster_name='xtremio_cluster'
+- san_login='xtremio_user'
+- san_password='xtremio_password'
+
+Here is a mock of the UI for the Storage Domain:
+
+![](/images/wiki/CinderLibStorageDomain.png)
+
+### High level flow
+
+Cinderlib needs to persist volumes metadata in order to perform some of the operation. The persistency of the volumes will be stored in a Postgres DB in a new database.
+
+The Cinderlib APIs will be invoked by the Engine, along with the DB parameters and the Drivers parameters.
+The Engine needs access to the Storage Management API network.
+
+Provisioning operations are done only on the Engine side.
+
+#### Create Storage Domain
+The user configures a new Storage Domain with all the needed parameters for the Cinder driver:
+  - The Engine calls a Cinderlib API to test the given parameters.
+  - If the call to Cinderlib API is successful, the Engine stores the data in the DB.
+
+#### Create Disk
+
+The user creates a disk on the storage domain:
+  - The Engine calls a Cinderlib API with the driver backend parameters, the volume id, the volume size and the DB URL.
+  - If the call to Cinderlib API is successful, the Engine creates the Disk in the DB.
+
+#### Delete Disk
+
+The user delete a disk on the storage domain:
+  - The Engine calls a Cinderlib API with the driver backend parameters, the volume id, the volume size and the DB URL to delete the volume.
+  - If the call to Cinderlib API is successful, the Engine deletes the disk in the DB
+
```

Comment:
we can add also the extend disk  operation

Reply:
Done